refactor(core): temporarily mark subscribe methods as deprecated

crisbeto opened the pull request:
The `subscribe` methods on `ModelSignal` and `OutputEmitter` were marked as `@internal` which will break when the TCB needs to reference them. These changes make them `@deprecated` temporarily so we can address the properly later.

devversion commented on packages/core/src/authoring/output.ts:
```diff
@@ -24,8 +24,9 @@ import {EventEmitter} from '../event_emitter';
+  // TODO(crisbeto): mark this as @internal
   // TODO: Consider exposing `subscribe` for dynamically created components.
-  /** @internal */
+  /** @deprecated Do not use, will be removed. */
```
You can keep this. Right now `output()` is not exposed and we are actually not sure if we want to expose it or not.

crisbeto committed:
refactor(core): temporarily mark subscribe methods as deprecated

The `subscribe` methods on `ModelSignal` and `OutputEmitter` were marked as `@internal` which will break when the TCB needs to reference them. These changes make them `@deprecated` temporarily so we can address the properly later.

crisbeto commented:
Caretaker note: there's one flaky target that failed.

jessicajaniuk commented:
This PR was merged into the repository by commit adfc3f0f95acf327d8c2ba80fb929f72a8226da4.
